src/taskgraph/decision.py

```diff
@@ -236,6 +244,26 @@ def get_decision_parameters(graph_config, options):
     return result
 
 
+def _determine_more_accurate_base_ref(repo, candidate_base_ref, head_ref, base_rev):
+    base_ref = candidate_base_ref
+
+    if not candidate_base_ref:
+        base_ref = repo.default_branch
+    elif candidate_base_ref == head_ref and base_rev == Repository.NULL_REVISION:
+        logger.warning(
+            "base_ref and head_ref are identical but base_rev equals the null revision. "
+            "This is a new branch but Github didn't identify its actual base."
```

Comment:
aside: I wonder if we should stop assuming Github in Taskgraph :). This is fine for now though, there are plenty bigger Github assumptions in Taskgraph.